src/core/event-handler.js

```diff
@@ -165,33 +175,46 @@ class EventHandler {
 
         if (!name) {
             // remove all events
+            for (const callbacks of this._callbacks.values()) {
+                for (let i = 0; i < callbacks.length; i++) {
+                    callbacks[i].destroy();
+                }
+            }
             this._callbacks.clear();
         } else if (!callback) {
             // remove all events of a specific name
-            if (this._callbacks.has(name))
+            const callbacks = this._callbacks.get(name);
+            if (callbacks) {
+                for (let i = 0; i < callbacks.length; i++) {
+                    callbacks[i].destroy();
+                }
                 this._callbacks.delete(name);
+            }
         } else {
-            const events = this._callbacks.get(name);
-            if (!events)
+            const callbacks = this._callbacks.get(name);
+            if (!callbacks)
                 return this;
 
-            let count = events.length;
+            let count = callbacks.length;
 
             for (let i = 0; i < count; i++) {
                 // remove all events with a specific name and a callback
-                if (events[i].callback !== callback)
+                if (callbacks[i].callback !== callback)
                     continue;
 
                 // could be a specific scope as well
-                if (scope && events[i].scope !== scope)
+                if (scope && callbacks[i].scope !== scope)
                     continue;
 
-                events[i--] = events[--count];
+                callbacks[i].destroy();
+                // potential issue with such way of removing items from an array,
+                // as it changes the order of event handlers
+                callbacks[i--] = callbacks[--count];
```

Comment:
Thank you for pointing this out, what if we just fix it since we are already contemplating these issues? A simple implementation:

```js
/**
 * const arr = [10, 20, 30, 40, 50, 60, 1, 2, 3, 70, 5, 4, 80];
 * filterInPlace(arr, _ => _ > 10);
 * console.log(arr); // Outputs: [20, 30, 40, 50, 60, 70, 80]
 * @param {T[]} arr 
 * @param {(T) => boolean} predicate 
 * @template T
 */
function filterInPlace(arr, predicate) {
  const n = arr.length;
  let iNew = 0;
  for (let i = 0; i < n; i++) {
    if (predicate(arr[i])) {
      arr[iNew++] = arr[i];
    }
  }
  arr.length = iNew;
}
```

The difference would be that the `callbacks` array would only "drop" to final length once every element was iterated. But since this is an internal property, it shouldn't matter? I mean... changing order doesn't matter either :thinking: :sweat_smile:

Reply:
Feeling unsure about this now, because we could just add an `EventHandle` while iterating probably, which makes order shuffling probably less of an issue. I cannot come up with an algorithm that fixes the length immediatly while filtering without breaking the order...

Reply:
This approach just works:
````js
let items = [ 1, 2, 3, 4, 5, 6, 7, 8, 9, 10 ];
for(let i = 0; i < items.length; i++) {
    if (items[i] === 3 || items[i] === 1 || items[i] === 5) {
        items.splice(i, 1);
    }
}
````